src/index.ts

```diff
@@ -18,6 +24,27 @@ export interface JsonFetchOptions extends Omit<RequestInit, 'body'> {
   retry?: Parameters<typeof promiseRetry>[0];
   timeout?: number;
   expectedStatuses?: Array<number>;
+  onRequestStart?: ({
+    agent,
+    body,
+    shouldRetry,
+    retry,
+    timeout,
+    expectedStatuses,
+    url,
+    retryCount,
+  }: OnRequestOptions) => void;
```

Comment:
I'm not sure it makes a ton of sense for the type for `onRequestStart` and `onRequestEnd` to be the same. There will never be `responseOrError` on `onRequestStart` but there will always be on `onRequestEnd`. Let's keep them separate?